parquet-tools/src/test/java/org/apache/parquet/tools/read/TestSimpleRecordConverter.java

```diff
@@ -0,0 +1,147 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.tools.read;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.column.ParquetProperties;
+import org.apache.parquet.example.data.Group;
+import org.apache.parquet.example.data.simple.SimpleGroupFactory;
+import org.apache.parquet.hadoop.ParquetReader;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.example.GroupWriteSupport;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+
+public class TestSimpleRecordConverter {
+
+  private static final String INT32_FIELD = "int32_field";
+  private static final String INT64_FIELD = "int64_field";
+  private static final String FLOAT_FIELD = "float_field";
+  private static final String DOUBLE_FIELD = "double_field";
+  private static final String BINARY_FIELD = "binary_field";
+  private static final String FIXED_LEN_BYTE_ARRAY_FIELD = "flba_field";
+
+
+  private File testFile;
+
+  @Test
+  public void testConverter() throws IOException {
+    ParquetReader<SimpleRecord> reader =
+      ParquetReader.builder(new SimpleReadSupport(), new Path(this.testFile.getAbsolutePath())).build();
+    for (SimpleRecord record = reader.read(); record != null; record = reader.read()) {
+      for (SimpleRecord.NameValue value : record.getValues()) {
+        switch(value.getName()) {
+          case INT32_FIELD:
+            Assert.assertEquals(32, value.getValue());
+            break;
+          case INT64_FIELD:
+            Assert.assertEquals(64L, value.getValue());
+            break;
+          case FLOAT_FIELD:
+            Assert.assertEquals(1.0f, value.getValue());
+            break;
+          case DOUBLE_FIELD:
+            Assert.assertEquals(2.0d, value.getValue());
+            break;
+          case BINARY_FIELD:
+            Assert.assertArrayEquals("foobar".getBytes(), (byte[])value.getValue());
+            break;
+          case FIXED_LEN_BYTE_ARRAY_FIELD:
+            Assert.assertArrayEquals(new byte[]{ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 }, (byte[])value.getValue());
+            break;
+        }
+      }
+    }
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    this.testFile = createTempFile();
+    MessageType schema = createSchema();
+    write(schema, testFile);
+  }
+
+  @After
+  public void tearDown() {
+    this.testFile.delete();
```

Comment:
When using the junit [TemporaryFolder](https://garygregory.wordpress.com/2010/01/20/junit-tip-use-rules-to-manage-temporary-files-and-folders/) we don't need this anymore.